src/libs/actions/IOU.js

```diff
@@ -789,6 +789,89 @@ function splitBillAndOpenReport(participants, currentUserLogin, currentUserAccou
     Report.notifyNewAction(groupData.chatReportID, currentUserAccountID);
 }
 
+/**
+ * @param {String} transactionID
+ * @param {Number} transactionThreadReportID
+ * @param {Object} transactionChanges
+ */
+function editMoneyRequest(transactionID, transactionThreadReportID, transactionChanges) {
+    // STEP 1: Get all collections we're updating
+    const transactionThread = allReports[`${ONYXKEYS.COLLECTION.REPORT}${transactionThreadReportID}`];
+    const transaction = allTransactions[`${ONYXKEYS.COLLECTION.TRANSACTION}${transactionID}`];
+    const iouReport = allReports[`${ONYXKEYS.COLLECTION.REPORT}${transactionThread.parentReportID}`];
+
+    // STEP 2: Build new modified expense report action.
+    const updatedReportAction = ReportUtils.buildOptimisticModifiedExpenseReportAction(transactionThread, transaction, transactionChanges);
+    const updatedTransaction = TransactionUtils.getUpdatedTransaction(transaction, transactionChanges);
+    // STEP 3: Compute the IOU total and update the report preview message so LHN amount owed is correct
```

Comment:
This is added to the follow up as it can be quite tricky with the expense (negative total) and with IOUs potentially changing the owner, I think it will be cleaner to do it in a separate PR